src/envoy/http/jwt_auth/jwt.cc

```diff
@@ -333,8 +333,12 @@ bool Verifier::VerifySignatureRSA(EVP_PKEY *key, const EVP_MD *md,
                                   size_t signed_data_len) {
   bssl::UniquePtr<EVP_MD_CTX> md_ctx(EVP_MD_CTX_create());
```

Comment:
I think there is a memory leak here.
I don't thee when the allocated value returned by EVP_MD_CTX_create() is free()'d.

Reply:
It is assigned to a smart_ptr bssl::UniquePtr.  It should free the buffer.

Reply:
Should we call `EVP_MD_CTX_free()` explicitly in this case?

Reply:
How does smart_ptr knows which function to call to free it?

Reply:
It would work if and only if `EVP_MD_CTX_create()` would not be a structure with other fields allocated.
But from the actual openSSL code, you can see there is a need to cleanup fields for the opaque structure pointed by `EVP_MD_CTX`, noticeably `ctx->md_data`, c.f.:

I'm good at C, not C++, but I can't see how smart_ptr would figure out how to free fields of an opaque structure. (opaque meaning, the structure type name appears in .h, but its fields are only in .c, so the scope prevents you to know the content of it, it's basically like an object with only private elements, not calling its memory-releasing/destruction function will create a memory leak).

```c
void EVP_MD_CTX_destroy(EVP_MD_CTX *ctx)
	{
	EVP_MD_CTX_cleanup(ctx);
	OPENSSL_free(ctx);
	}

/* This call frees resources associated with the context */
int EVP_MD_CTX_cleanup(EVP_MD_CTX *ctx)
	{
	/* Don't assume ctx->md_data was cleaned in EVP_Digest_Final,
	 * because sometimes only copies of the context are ever finalised.
	 */
	if (ctx->digest && ctx->digest->cleanup
	    && !EVP_MD_CTX_test_flags(ctx,EVP_MD_CTX_FLAG_CLEANED))
		ctx->digest->cleanup(ctx);
	if (ctx->digest && ctx->digest->ctx_size && ctx->md_data
	    && !EVP_MD_CTX_test_flags(ctx, EVP_MD_CTX_FLAG_REUSE))
		{
		OPENSSL_cleanse(ctx->md_data,ctx->digest->ctx_size);
		OPENSSL_free(ctx->md_data);
		}
	if (ctx->pctx)
		EVP_PKEY_CTX_free(ctx->pctx);
#ifndef OPENSSL_NO_ENGINE
	if(ctx->engine)
		/* The EVP_MD we used belongs to an ENGINE, release the
		 * functional reference we held for this reason. */
		ENGINE_finish(ctx->engine);
#endif
	memset(ctx,'\0',sizeof *ctx);

	return 1;
	}
```